x-pack/plugins/apm/server/routes/apm_routes/register_apm_server_routes.ts

```diff
@@ -70,130 +75,157 @@ export function registerRoutes({
-          if (!options.disableTelemetry && telemetryUsageCounter) {
-            telemetryUsageCounter.incrementCounter({
-              counterName: `${method.toUpperCase()} ${pathname}`,
-              counterType: 'success',
-            });
-          }
-
-          return response.ok({ body });
-        } catch (error) {
-          logger.error(error);
-
-          if (!options.disableTelemetry && telemetryUsageCounter) {
-            telemetryUsageCounter.incrementCounter({
-              counterName: `${method.toUpperCase()} ${pathname}`,
-              counterType: 'error',
-            });
-          }
-          const opts = {
-            statusCode: 500,
-            body: {
-              message: error.message,
-              attributes: {
-                _inspect: inspectableEsQueriesMap.get(request),
-              },
+        return response.ok({ body });
+      } catch (error) {
+        logger.error(error);
+
+        if (!options.disableTelemetry && telemetryUsageCounter) {
+          telemetryUsageCounter.incrementCounter({
+            counterName: `${method.toUpperCase()} ${pathname}`,
+            counterType: 'error',
+          });
+        }
+        const opts = {
+          statusCode: 500,
+          body: {
+            message: error.message,
+            attributes: {
+              _inspect: inspectableEsQueriesMap.get(request),
             },
-          };
+          },
+        };
 
-          if (error instanceof errors.RequestAbortedError) {
-            return response.custom(merge(opts, CLIENT_CLOSED_REQUEST));
-          }
+        if (error instanceof errors.RequestAbortedError) {
+          return response.custom(merge(opts, CLIENT_CLOSED_REQUEST));
+        }
 
-          if (Boom.isBoom(error)) {
-            opts.statusCode = error.output.statusCode;
-          }
+        if (Boom.isBoom(error)) {
+          opts.statusCode = error.output.statusCode;
+        }
 
-          // capture error with APM node agent
-          apm.captureError(error);
+        // capture error with APM node agent
+        apm.captureError(error);
 
-          return response.custom(opts);
-        } finally {
-          // cleanup
-          inspectableEsQueriesMap.delete(request);
-        }
+        return response.custom(opts);
+      } finally {
+        // cleanup
+        inspectableEsQueriesMap.delete(request);
       }
-    );
+    };
+
+    if (!version) {
+      (
+        router[method] as RouteRegistrar<
+          typeof method,
+          ApmPluginRequestHandlerContext
+        >
+      )(
+        {
+          path: pathname,
+          options,
+          validate: routeValidationObject,
+        },
+        wrappedHandler
+      );
+    } else {
+      (
+        router.versioned[method] as VersionedRouteRegistrar<
+          typeof method,
+          ApmPluginRequestHandlerContext
+        >
+      )({
+        path: pathname,
+        access: pathname.includes('/internal/apm') ? 'internal' : 'public',
```

Comment:
Is seems that we rely more and more to the naming scheme to distinguish internal / public APIs, wondering if we can improve this. It's also mentioned here https://github.com/elastic/kibana/issues/151940

Reply:
What do you think needs improvement? You'd rather have an explicit `access` prop on all routes?

Reply:
This might be an idea but not too precious about it